docs/source/advanced_patterns.rst

```diff
@@ -0,0 +1,107 @@
+.. _advanced_communication_patterns:
+
+
+*******************************
+Advanced communication patterns
+*******************************
+
+Some advanced communication patterns are also implemented in osBrain. They are
+called channels, rather than sockets, as they are formed with multiple sockets.
+
+
+.. index:: async-req-rep
+.. _async_req_rep:
+
+Asynchronous Request-Reply channel
+==================================
+
+An asynchronous request-reply channel is very similar to a normal request-reply
+pattern, only in this case we want to process the reply asynchronously (i.e.:
+assign a handler for this reply and forget about it until it is received).
+
+This means that, when using an asynchronous request-reply channel, a handler
+must be specified not only for the replier but also for the requester.
```

Comment:
Maybe '... must be specified for both the replier and the requester' would sound simpler.